src/wp-admin/theme-editor.php

```diff
@@ -212,19 +212,45 @@
 	);
 }
 
-if ( preg_match( '/\.css$/', $file ) && ! wp_is_block_theme() && current_user_can( 'customize' ) ) {
-	$message = '<p><strong>' . __( 'Did you know?' ) . '</strong></p><p>' . sprintf(
-		/* translators: %s: Link to Custom CSS section in the Customizer. */
-		__( 'There is no need to change your CSS here &mdash; you can edit and live preview CSS changes in the <a href="%s">built-in CSS editor</a>.' ),
-		esc_url( add_query_arg( 'autofocus[section]', 'custom_css', admin_url( 'customize.php' ) ) )
-	) . '</p>';
-	wp_admin_notice(
-		$message,
-		array(
-			'type' => 'info',
-			'id'   => 'message',
-		)
-	);
+if ( preg_match( '/\.css$/', $file ) ) {
+	if ( ! wp_is_block_theme() && current_user_can( 'customize' ) ) {
+		$message = '<p><strong>' . __( 'Did you know?' ) . '</strong></p><p>' . sprintf(
+			/* translators: %s: Link to add custom CSS section in either the Customizer (classic themes) or Site Editor (block themes). */
+			__( 'There is no need to change your CSS here &mdash; you can edit and live preview CSS changes in the <a href="%s">built-in CSS editor</a>.' ),
+			esc_url( add_query_arg( 'autofocus[section]', 'custom_css', admin_url( 'customize.php' ) ) )
+		) . '</p>';
+		wp_admin_notice(
+			$message,
+			array(
+				'type' => 'info',
+				'id'   => 'message',
+			)
+		);
+	} elseif ( wp_is_block_theme() && current_user_can( 'edit_theme_options' ) ) {
+		$message = '<p><strong>' . __( 'Did you know?' ) . '</strong></p><p>' . sprintf(
+			/* translators: %s: Link to add custom CSS section in either the Customizer (classic themes) or Site Editor (block themes). */
+			__( 'There is no need to change your CSS here &mdash; you can edit and live preview CSS changes in the <a href="%s">built-in CSS editor</a>.' ),
+			esc_url( admin_url( 'site-editor.php?p=%2Fstyles&section=%2Fcss' ) )
+		) . '</p>';
+		wp_admin_notice(
+			$message,
+			array(
+				'type' => 'info',
+				'id'   => 'message',
+			)
+		);
+	}
+	if ( file_exists( preg_replace( '/\.css$/', '.min.css', $file ) ) ) {
```

Comment:
@westonruter In the development version, `SCRIPT_DEBUG` is set to `true`, so this notice does not makes sense since it loads CSS from the unminified `.css` file. However, in production, `SCRIPT_DEBUG` is `false`, meaning that CSS won’t be applied as it load `.min.css` file.

This creates a bit of a trade-off not sure if we actually need this notice in that case 🤔

Reply:
Got it. Thanks for clarification.